word_counter.rb

```diff
@@ -0,0 +1,79 @@
+require 'pragmatic_segmenter'
+# how do you include a gem such as this so that the program will run on a given computer?
+
+def word_counter
+	# Lists words, their frequency, and location based on sentence # from given text doc.
+	# The way the file is read, it could freeze the host if file given is too big.
+	# Will not work with languages other than English
+	#
+	# Started 9/16/2015
+	# Updated 2/14/2016
+	# Author: Christopher Gabriel Manna
+	# find a way to turn this into a class
+	
+	# setting new values
+	h = Hash.new(0)
+	i = 0
+	alph = "a".ord
+	pot_sentences = []
+	words = []
+	
+
+	file_name = ARGV[0]
+	if (File.extname(file_name).to_s == ".txt" && File.exist?(file_name))
```

Comment:
`&&` is great because if won't run the second argument if the first evaluates to false. You can use this property to make this line slightly more efficient by placing `File.exist?(file_name)` first. That way, you will never even check the file extension if the file doesn't exist.